src/backend/utils/misc/queryenvironment.c

```diff
@@ -748,6 +794,23 @@ static bool _ENR_tuple_operation(Relation catalog_rel, HeapTuple tup, ENRTupleOp
 					list_ptr = &enr->md.cattups[ENR_CATTUP_CLASS];
 					lc = list_head(enr->md.cattups[ENR_CATTUP_CLASS]);
 					ret = true;
+
+					/*
+					 * All of the below asserts should hold true for TSQL temp tables.
+					 *
+					 * This helps ensure that we don't have any dependencies pointing to
+					 * non-ENR catalogs.
+					 */
+					if ((op == ENR_OP_ADD || op == ENR_OP_DROP) && HeapTupleIsValid(tup))
+					{
+						Form_pg_class classForm = (Form_pg_class) GETSTRUCT(tup);
+
+						Assert(!classForm->relisshared);
+						Assert(classForm->relpersistence == RELPERSISTENCE_TEMP);
+						Assert(!classForm->relhastriggers);
+						Assert(!classForm->relrowsecurity);
+						Assert(!classForm->relispartition);
```

Comment:
Are Asserts enabled in prod? If not, im curious how we would catch the issues in prod?